lib/Arch/X86/Arch.cpp

```diff
@@ -788,6 +788,66 @@ static void DecodeOperand(Instruction &inst, const xed_decoded_inst_t *xedd,
   }
 }
 
+static uint64_t BytesToCallBits(const uint8_t *bytes) {
+  // The pattern we're looking for is 5 bytes long.
+  uint64_t bits = 0;
+  bits = (bits << 8) | static_cast<uint32_t>(bytes[0]);
+  bits = (bits << 8) | static_cast<uint32_t>(bytes[1]);
+  bits = (bits << 8) | static_cast<uint32_t>(bytes[2]);
+  bits = (bits << 8) | static_cast<uint32_t>(bytes[3]);
+  bits = (bits << 8) | static_cast<uint32_t>(bytes[4]);
+  return bits;
+}
+
+static bool TryDecodeIdioms(Instruction &inst) {
+  // Check for CALL+POP idiom used to retrieve PC.
+  if (inst.bytes.size() == 6) {
```

Comment:
Early return here - if we know already that is not `6` just return `false` - it will simplify the code (because you do not need to scroll all way down for that `return false` and spare an indent.

Reply:
Do we expect to add more idioms? That's the reason I wrote it like this (and named the function in a more generic way).

Reply:
That's a valid point, but I would consider extracting this idiom as separate function then? We don't want to have them all inlined.

Reply:
Start with `inst.bytes.size() >= 6`, as you might be given up to 15 bytes.

Reply:
Good idea.

[5a4470b](https://github.com/lifting-bits/remill/pull/610/commits/5a4470b69bb3023fafdcbd1c61b0268d466243a8)